### PR TITLE
REF: separate out invalid ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
We moved ops.py to `ops.__init__` a while back, still need to get the bulk of it out of `__init__`.  This separates out templated invalid operations, which are the main things that outside modules import  (this helps us move towards getting rid of the `import pandas as pd` in the main file)

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Accidentally just pushed what was supposed to be a separate branch.  Since I already have too many PRs and this PR was small to begin with, gonna just roll with it.

The new commit replaces usages of dispatch_to_index_op with usages of dispatch_to_extension_op.  There is one usage of dispatch_to_index_op which is not yet removed, pending the fixing of a bug in NaT just identified.

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -57,6 +57,7 @@
     _make_flex_doc,
     _op_descriptions,
 )
```
Comment (jreback) [inline pandas/core/ops/__init__.py]: use absolute imports

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: I think we're holding off on these until the `import pandas as pd` is removed from this file

Comment (jreback) [reply pandas/core/ops/__init__.py]: that's orthogonal. these should not have been like this in the first place.

Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: updated

Comment (jreback) [reply pandas/core/ops/__init__.py]: thxs, ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
Rebased and green

**Pull request merged**